src/host/main.cpp

```diff
@@ -458,6 +458,21 @@ int main(int argc, char** argv)
     }
 
     LOG_INFO_FMT("Initialising enclave: enclave_create_node");
+    std::atomic<bool> ecall_completed = false;
+    auto flush_outbound = [&]() {
+      while (true)
```

Comment:
while (!ecall_completed) ? Always a bit nervous when I see a while(true);

Reply:
Sure, I'll re-write this as a `do-while` (I'd prefer that to a `while`, so that we clearly try to call `read_all` once _after_ `ecall_completed` has been set. We can't perfectly sync, but that's fine (anything that gets left on the boundary will get flushed eventually). This pattern just does its best to flush as early as possible.